cli/src/modules/message.rs

```diff
@@ -0,0 +1,158 @@
+    }
+
+    async fn handle(self: Arc<Self>, ctx: &Arc<dyn Context>, argv: Vec<String>, cmd: &str) -> cli::Result<()> {
+        let ctx = ctx.clone().downcast_arc::<KaspaCli>()?;
+        self.main(ctx, argv, cmd).await.map_err(|e| e.into())
+    }
+}
+
+impl Message {
+    async fn main(self: Arc<Self>, ctx: Arc<KaspaCli>, argv: Vec<String>, _cmd: &str) -> Result<()> {
+        if argv.is_empty() {
+            return self.display_help(ctx, argv).await;
+        }
+
+        match argv.get(0).unwrap().as_str() {
+            "sign" => {
+                if argv.len() < 2 {
+                    return self.display_help(ctx, argv).await;
+                }
+
+                let kaspa_address = argv.get(1).unwrap().as_str();
+                let asked_message = ctx.term().ask(false, "Message: ").await?;
+                let message = asked_message.as_str();
+
+                self.sign(ctx, kaspa_address, message).await?;
+            }
+            "verify" => {
+                if argv.len() < 3 {
+                    return self.display_help(ctx, argv).await;
+                }
+                let kaspa_address = argv.get(1).unwrap().as_str();
+                let signature = argv.get(2).unwrap().as_str();
+                let asked_message = ctx.term().ask(false, "Message: ").await?;
+                let message = asked_message.as_str();
+
+                self.verify(ctx, kaspa_address, signature, message).await?;
+            }
+            v => {
+                tprintln!(ctx, "unknown command: '{v}'\r\n");
+                return self.display_help(ctx, argv).await;
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn display_help(self: Arc<Self>, ctx: Arc<KaspaCli>, _argv: Vec<String>) -> Result<()> {
+        ctx.term().help(
+            &[
+                ("sign <kaspa_address>", "Sign a message with the private key that matches the given address. Prompts for message."),
+                (
+                    "verify <kaspa_address> <signature>",
+                    "Verify the signature against the message and kaspa_address. Prompts for message.",
+                ),
+            ],
+            None,
+        )?;
+
+        Ok(())
+    }
+
+    async fn sign(self: Arc<Self>, ctx: Arc<KaspaCli>, kaspa_address: &str, message: &str) -> Result<()> {
+        let kaspa_address = Address::try_from(kaspa_address)?;
+        if kaspa_address.version != Version::PubKey {
+            return Err(Error::custom("Address not supported for message signing. Only supports PubKey addresses"));
+        }
+
+        let pm = PersonalMessage(message);
+        let privkey = self.get_address_private_key(&ctx, kaspa_address).await?;
+
+        let sig_result = sign_message(&pm, &privkey);
+
+        match sig_result {
+            Ok(signature) => {
+                let sig_hex = faster_hex::hex_string(signature.as_slice());
+                tprintln!(ctx, "Signature: {}", sig_hex);
+            }
```

Comment:
```
Ok(())
```
Then you will be able to remove `return` word in the second condition branch and 'Ok(())' below